core/src/actors/paths.rs

```diff
@@ -292,22 +305,42 @@ impl ActorPath {
         from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env))
     }
 
-    /// Same as [tell](ActorPath::tell), but serialises eagerly
-    pub fn tell_ser<S, B, E>(&self, m: B, from: &S) -> Result<(), E>
-    where
-        S: ActorSource,
-        B: TryInto<Serialised, Error = E>,
+    /// Same as [tell](ActorPath::tell), but serialises eagerly into a Pooled buffer (pre-allocated and bounded)
+    pub fn tell_serialised<CD, B>(&self, m: B, from: &CD) -> Result<(), SerError>
+        where
+            CD: ComponentDefinition + Sized + 'static,
+            B: Serialisable + 'static,
     {
-        let msg: Serialised = m.try_into()?;
-        let src = from.path_resolvable();
-        let dst = self.clone();
-        let env = DispatchEnvelope::Msg {
-            src,
-            dst,
-            msg: DispatchData::Eager(msg),
-        };
-        from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env));
-        Ok(())
+        if self.protocol() == Transport::LOCAL {
+            // No need to serialize!
+            self.tell(m, from);
+            return Ok(());
+        }
+        //let mut buf = encode_buffer.deref_mut();
+        let mut buf_ref = from.ctx().get_buffer().borrow_mut();
+        let buf = buf_ref.deref_mut();
+        // Reserve space for the header:
+        buf.pad(FRAME_HEAD_LEN as usize);
+
+        from.actor_path().serialise(buf)?; // src
+        self.clone().serialise(buf)?; // dst
+        buf.put_ser_id(m.ser_id()); // ser_id
+        Serialisable::serialise(&m, buf)?; // data
+
+        if let Some(mut chunk_lease) = buf.get_chunk() {
+            // The length which a FrameHead tells does not include the size of the FrameHead itself
+            let len = chunk_lease.remaining() - FRAME_HEAD_LEN as usize;
+            chunk_lease.insert_head(FrameHead::new(FrameType::Data, len));
+            let env = DispatchEnvelope::Msg {
+                src: from.path_resolvable(),
+                dst: self.clone(),
+                msg: DispatchData::Serialised((chunk_lease, m.ser_id())),
+            };
+            from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env));
+            Ok(())
+        } else {
+            panic!("failed to get buffer!");
```

Comment:
Probably not a good idea to panic here. A SerErr perhaps?

Reply:
Yes, not being able to get a buffer should be a special variant of `SerErr` (since it may require special handling later). But Adam and I already discussed earlier that this PR won't necessary have all the proper error handling, yet.
Just leave a TODO note there or something (maybe one tied to an Issue, so we don't forget).